src/Nethermind/Nethermind.JsonRpc/Modules/Admin/PeerInfo.cs

```diff
@@ -43,13 +45,13 @@ public PeerInfo(Peer peer, bool includeDetails)
             IsBootnode = peer.Node.IsBootnode;
             IsStatic = peer.Node.IsStatic;
             Enode = peer.Node.ToString(Node.Format.ENode);
+            Inbound = peer.InSession is not null && peer.InSession.Direction == ConnectionDirection.In; // assumes one of InSession/Outsession is always set.?
 
-            if (includeDetails)
-            {
-                ClientType = peer.Node.ClientType.ToString();
-                EthDetails = peer.Node.EthDetails;
-                LastSignal = (peer.InSession ?? peer.OutSession)?.LastPingUtc.ToString(CultureInfo.InvariantCulture);
-            }
+            if (!includeDetails) return;
```

Comment:
personal: I dislike jumping out from methods in the middle (it is okayish at the begining).